core/src/basic_block.rs

```diff
@@ -34,6 +84,20 @@ pub struct BasicBlock {
     pub mnemonics: Vec<Mnemonic>,
 }
 
+// this gets us iteration over mnemonics from a basic block for free
```

Comment:
I'm not sure about this. BasicBlock isn't just a smart pointer to a Vec<Mnemonic>. I think it's better to use a separate function instead of overloading an operator.